### PR TITLE
Revamps Conveyor Belt Construction

### PR DESCRIPTION
**Pull request by clusterfack**
Conveyor belt assemblies now use click drag system to be placed onto the ground, from which they get their direction. Conveyor belt assemblies now only require the four metal plates to be added for them to function instead of circuitboards.

### CODE REVIEW

File: vgstation13.dme [hunk 1]
```diff
@@ -2044,7 +2044,7 @@
 #include "interface\web\interface.dms"
 #include "maprendering\maprendering.dm"
 #include "maps\_map.dm"
-#include "maps\tgstation.dm"
+#include "maps\test_box.dm"
```
Comment (JustSumBody) [inline vgstation13.dme]: uh oh

**Comment by clusterfack** [comment]
I could use a different sprite for the in hand conveyor stack. Since they look identical to the placed versions it gets very confusion very quickly.

**Comment by clusterfack** [comment]
Alright this is complete

**Comment by 9600bauds** [comment]
Directions are fucked up when trying to set them via multitool menu, which wasn't updated
This removes copying direction when cloning from another conveyor belt, which is a huge hassle and conflicts with the info on the multitool configuration screen (please use a multitool on a conveyor belt to see what I mean)
Deconstructing a conveyor belt still pries out the circuitboard, does not return any metal
Kinda difficult to try to do long straight sections of track because when moving away from the track to continue building it, you turn away from it: have to turn back to face it every square of the stretch.
Not sure if this is intended but the conveyor belt assemblies are unanchored

Circuitboard still exists in circuit imprinter, soldering tool etc. despite not being used anymore
Impossible to deconstruct conveyor belt assemblies back into metal
Examining a stack of conveyor belt assemblies says "There are 18 s in the stack.". Where does that rogue "s" come from?

File: code/modules/recycling/conveyor_assembly.dm [hunk 2]
```diff
@@ -2,82 +2,101 @@
...
-					returnToPool(O)
-				CB.component_parts = list()
-				circuit.forceMove(null) //I swear this is what happens for constructable frames
-				CB.component_parts += circuit
-				circuit = null
+	else if(istype(P, /obj/item/stack/sheet/metal))
+		var/obj/item/stack/S = P
+		if(S.amount > 4)
+			playsound(get_turf(src), 'sound/items/Ratchet.ogg', 75, 1)
+			if(do_after(user, src, 30) && S.amount > 4)
```
Comment (9600bauds) [inline code/modules/recycling/conveyor_assembly.dm:114]: Would be nice to have a warning here if you don't have enough sheets in your held stack

**Comment by clusterfack** [comment]
I removed copying directions explicitly, because it makes copying frequencies more of a task to longer strands of belts that aren't always going the same direction, and because my system makes specifying direction of belts EXTREMELY easy such that you simply will not need to use a multitool to specify it.

>you turn away from it: have to turn back to face it every square of the stretch.

Not sure what you mean by this?


>Not sure if this is intended but the conveyor belt assemblies are unanchored

If you mean the stacks, intended, if you mean the structure. It should inherit anchored = 1 from its parent.

>Circuitboard still exists in circuit imprinter, soldering tool etc. despite not being used anymore
Impossible to deconstruct conveyor belt assemblies back into metal
Examining a stack of conveyor belt assemblies says "There are 18 s in the stack.". Where does that rogue "s" come from?

These are some good oversights, thanks

**Comment by 9600bauds** [comment]
The structure is unanchored when building it and before adding sheets, yes

dreamseeker_2016-10-21_00-32-24

**Comment by clusterfack** [comment]
I'll change that text then? :shrug:

Changing directions after the fact just isn't really that important with this.

**Comment by 9600bauds** [comment]
The functionality to copy only the frequency and ID tag already exists, why are you removing the ability to copy direction if you want to, too?

>Changing directions after the fact just isn't really that important with this.

Would be nice if it worked anyways

**Comment by clusterfack** [comment]
Working is one thing but it decreases multitool cloning usability doing it that way.

**Comment by 9600bauds** [comment]
You have two different functionalities depending on what you want to do
Removing one of them does not add usability

**Comment by 9600bauds** [comment]
About the "face away" thing I said earlier, seems like corners and the straight tracks are kind of inverted from eachother which results in kind of fucked up corners. The solution to this is to point the straight tracks kind of "away" from the corners.

**Comment by clusterfack** [comment]
Removing one of them that is now accomplished through another method more easily, so that the second which was made less useful thanks to a quirk of the first is now more useful.

**Comment by clusterfack** [comment]
>About the "face away" thing I said earlier, seems like corners and the straight tracks are kind of inverted from eachother which results in kind of fucked up corners. The solution to this is to point the straight tracks kind of "away" from the corners (you can't really point the corners "away" from the straight tracks).

Are they? Hm. I think I may have to take a second look at the conveyor dir selection proc.

**Comment by 9600bauds** [comment]
The direction cloning is useful if you are expanding the track to overwrite the old corners, or if, god forbid, mistakes are made when building it (like when I found out it got removed at all)
Once again: You can choose to not copy a direction by cloning from the switch (which you HAVE to have to use a conveyor belt at all), this is clearly explained in the multitool menu that you have to bring up to enable cloning at all, removing it because someone didn't read it is awful

**Comment by clusterfack** [comment]
I'm getting the feeling like the whole 8 directional conveyor belt should be scrapped entirely in favor of smoothing. Since there's always going to be this inherent ambiguity of direction as long as they exist.

@9600bauds If you're overwriting a corner, then you only need to use the multitool once, which defeats the whole purpose of *cloning* which is to be used several times. But being able to clone any track without needing to build a switch is a far superior QoL.

**Comment by 9600bauds** [comment]
You need to build a switch to use the belts at all...

**Comment by clusterfack** [comment]
And you'll never need to override direction normally...

**Comment by 9600bauds** [comment]
unless you're changing an already built track

**Comment by DrCelt** [comment]
Changing already built tracks is not some uncommon unheard of thing, Cluster.

**Comment by clusterfack** [comment]
Except that this cloning method only helps you create long straight line segments. Which if you're actually restructuring existing tracks is basically useless.

**Comment by clusterfack** [comment]
@DrCelt I'd rather add interaction with another tool to more easily change direction of conveyor belts than to muddy the multitools functionality

**Comment by 9600bauds** [comment]
Do that then

**Comment by 9600bauds** [comment]
Another reason why you could want direction copying is if you accidentally built one whole side of the straight conveyor belts on the opposite direction
Which I actually just did because I was facing the wrong way while laying down the assemblies

**Comment by clusterfack** [comment]
I'm thinking mousedrag while using crowbar is the best way here

**Comment by 9600bauds** [comment]
Could work, might be a bit annoying if you accidentally click and start disassembling though

**Comment by 9600bauds** [comment]
By dragging unanchored assemblies on top of eachother you can have multiple conveyor belts on the same tile and shits gets kinda cray

**Comment by clusterfack** [comment]
Yeah I didn't know the base structure was unanchored, actually it's still a little much to swallow? But I have it fixed in my next revision.

**Comment by 9600bauds** [comment]
You can't seem to stack them on top of eachother besides dragging so that's good

**Comment by Kurfursten** [comment]
This is neat.

**Comment by Iamgoofball** [comment]
just fucking make a tool that works like placing down pathing/track in a rollercoaster tycoon game

**Comment by DrCelt** [comment]
Go back to the shadows from whence you came, @Iamgoofball

**Comment by Iamgoofball** [comment]
you mean goofshadows

i removed the old shadow code it was too outdated

**Comment by clusterfack** [comment]
@Iamgoofball I already would have if byond's click handling code wasn't so shit.

**Comment by Kurfursten** [comment]
It could work like the RCL, though.

Totally out of scope of course. Perhaps a feature for another time.

**Comment by clusterfack** [comment]
This works 100x better than the rcl have you used it?

**Comment by Kurfursten** [comment]
No, I don't have time to download branches and test other people's code these days with grad school back on. Maybe after it gets merged I'll get a chance to play and see what you mean. With clickdrag I'm envisioning something similar to mousedrop mechanics but I am patient enough to wait and see what you mean later.

**Comment by clusterfack** [comment]
Fixed issues
[] Circuitboard still exists in circuit imprinter, soldering tool etc. despite not being used anymore
[] Impossible to deconstruct conveyor belt assemblies back into metal
[] Examining a stack of conveyor belt assemblies says "There are 18 s in the stack.". Where does that rogue "s" come from?

**Comment by clusterfack** [comment]
Alright this time I really do think it's done. Put in a few hours work to get it to properly find the correct diagonal direction and a pretty small amount of time to modify construction/deconstruction.

**Comment by 9600bauds** [comment]
Did not update wiki smh

**Comment by Probe1** [comment]
GET EM

**Comment by 9600bauds** [comment]
I did, I had to look this PR up to remember how it worked to answer an ahelp

**Pull request merged**